src/algorithms/aco.c

```diff
@@ -347,88 +354,54 @@ populacao *aco(populacao *population, int epoca_num, int population_num)
     }
 
     individuo *individuos = population->individuos;
-    int d = 10;
+ 
     DEBUG(printf("aco\n");)
-    // Allocate memory for the pheromone matrix
     double **pheromones = (double **)malloc(parameters.population_size * sizeof(double *));
     pheromones_candidates = (double **)malloc(parameters.num_candidates * sizeof(double *));
-    time_t time_init, time_now;
-    time(&time_init);
-    time(&time_now);
-
-    for (int i = 0; i < parameters.population_size; i++)
-    {
-        pheromones[i] = (double *)malloc(d * sizeof(double));
-        for (int j = 0; j < d; j++)
-        {
-            pheromones[i][j] = 1.0 / parameters.population_size;
-        }
-    }
-
-    for (int i = 0; i < parameters.num_candidates; i++)
-    {
-        pheromones_candidates[i] = (double *)malloc(d * sizeof(double));
-        for (int j = 0; j < d; j++)
-        {
-            pheromones_candidates[i][j] = 1.0 / parameters.num_candidates;
-        }
-    }
+    probability_interval = (domain **)malloc(parameters.population_size * sizeof(domain *));
+    candidates = (individuo *)malloc(parameters.num_candidates * sizeof(individuo));
```

Comment:
pelo que entendi essas variáveis globais não podem ser perdidas durante a troca de épocas, se sim, faz sentido ficar alocando elas aqui toda vez que a função `aco` for chamada pelos algoritmos de ilhas, o comentário vale tbm para: sigma, pheromones_candidates, probability_interval. Se for caso seria melhor alocar **todas (incluindo cada linha)** as posições da matriz globalmente tbm, pra que seja feito apenas uma vez

Agora se pode ser perdido, lembrar de desalocar a memória no fim da função `aco`